commit d2e34218068432ba9c312591067e53eda67cb68a
Author: Dominik Haumann <dhaumann@kde.org>
Date:   Wed Jun 12 22:30:32 2013 +0200

    fix crash in passive notification system
    
    @packagers / distributions: Better backport this to you 4.10.4 packages.
    
    FIXED-IN: 4.10.5
    BUG: 320302

diff --git a/part/view/katemessagewidget.cpp b/part/view/katemessagewidget.cpp
index d3c895f..6c227c0 100644
--- a/part/view/katemessagewidget.cpp
+++ b/part/view/katemessagewidget.cpp
@@ -225,11 +225,28 @@ void KateMessageWidget::startAutoHideTimer()
     return;
   }
 
+  // switching KateViews may result isVisible() == true and still m_messageList.size() == 0.
+  // The problem is that the hideEvent is never called for the KMessageWidget, if the
+  // parent widget is hidden. In that case, we 'miss' that the notification is gone...
+  if (m_messageList.size() == 0) {
+    m_hideAnimationRunning = false;
+    m_autoHideTimerRunning = false;
+    m_autoHideTime = -1;
+
+    if (isVisible()) {
+      m_hideAnimationRunning = true;
+      if (m_fadeEffect) {
+        m_fadeEffect->fadeOut();
+      } else {
+        m_messageWidget->animatedHide();
+      }
+    }
+    return;
+  }
+
   // remember that auto hide timer is running
   m_autoHideTimerRunning = true;
 
-  // the message must still still be valid
-  Q_ASSERT(m_messageList.size());
   KTextEditor::Message* message = m_messageList[0];
   QTimer::singleShot(m_autoHideTime == 0 ? (6*1000) : m_autoHideTime, message, SLOT(deleteLater()));
 }
